modal/sandbox.py

```diff
@@ -63,6 +64,23 @@ def _validate_exec_args(entrypoint_args: Sequence[str]) -> None:
         )
 
 
+@dataclass(frozen=True)
+class SandboxResourceUsage:
+    """Simple data structure storing resource usage stats of a Sandbox."""
+
+    # CPU usage is in core-nanoseconds. Using 1 core for 1 second is 1e9 core ns.
+    # Using 0.5 core for 1 second is 5e8 core ns.
+    cpu_core_nanosecs: int
+    # Memory usage is in gibibyte-nanoseconds. Using 1 GiB for 1 second is 1e9 gib ns.
+    # Using 0.5 GiB for 1 second is 5e8 gib ns.
+    mem_gib_nanosecs: int
+    # GPU type, e.g. "a10g". Combine with `gpu_nanosecs` to understand billable GPU spend.
```

Comment:
Currently we have a many-to-one mapping for some GPU types on configuration (e.g. we accept `"A10"` and `"A10g"`, plus we're case-insensitive, etc. How are we defining the schema for GPU type here? We've broken our own internal tooling recently with some migrations; are we now obligated to keep the current schema fixed moving forward?

Reply:
We have a canonical map in the backend, could just use that one. But I'm also fine not exposing it here.

Reply:
that makes sense to me in general but do sandboxes have gpu fallbacks? or can you only request one gpu type?